src/routes/register/+page.svelte

```diff
@@ -0,0 +1,349 @@
+<script>
+	export let data; // data from database
+
+	import Header from 'elements/header.svelte';
+	import SpacerHeader from 'elements/spacer/spacerHeader.svelte';
+	import Footer from 'elements/footer.svelte';
+
+	import Headline from 'elements/text/headline.svelte';
+	import HorizontalLine from 'elements/line/horizontalLine.svelte';
+	import Spacer from 'elements/spacer/spacer.svelte';
+	import MessageWrapper from 'elements/messanges/messageWrapper.svelte';
+	import ErrorMessage from 'elements/messanges/errorMessage.svelte';
+	import Input from 'elements/input/input.svelte';
+	import Textline from 'elements/text/textline.svelte';
+	import List from 'elements/list/list.svelte';
+	import ListElement from 'elements/list/listElement.svelte';
+	import Button from 'elements/input/button.svelte';
+	import Textblock from 'elements/text/textblock.svelte';
+
+	const headerMenu = [
+		['Start', '/'],
+		['Anmelden', '/login']
+	];
+	const footerMenu = [
+		['Anmelden', '/login'],
+		['Impressum', '/impressum']
+	];
+
+	let timerUsername = null;
+	let timerEmail = null;
+
+	let username = '';
+	let email = '';
+	let password_1 = '';
+	let password_2 = '';
+
+	let usernameMessage = '';
+	let emailMessage = '';
+	let passwordMessage = '';
+	let errorMessage = [''];
+
+	let registered = false;
+
+	function startTimerUsername() {
+		if (timerUsername !== null) {
+			clearTimeout(timerUsername);
+			timerUsername = null;
+		}
+		timerUsername = setTimeout(() => {
+			onUsernameChanged();
+		}, 1000);
+	}
+
+	function startTimerEmail() {
+		if (timerEmail !== null) {
+			clearTimeout(timerEmail);
+			timerEmail = null;
+		}
+		timerEmail = setTimeout(() => {
+			onEmailChanged();
+		}, 1000);
+	}
+
+	function samePassword() {
+		return password_1.trim() === password_2.trim();
+	}
+
+	async function onUsernameChanged() {
+		const reset = () => {
+			usernameMessage = '';
+		};
+
+		if (username.trim() === '') {
+			reset();
+			return;
+		}
+
+		const response = await fetch('api/account/username/exists?username=' + username);
```

Comment:
Shouldn't the username also be trimmed here?
```suggestion
		const response = await fetch('api/account/username/exists?username=' + username.trim());
```